packages/pieces/community/ing-australia/src/lib/actions/download-transactions.ts

```diff
@@ -0,0 +1,72 @@
+import { createAction, Property } from '@activepieces/pieces-framework';
+
+import {
+  HttpRequest,
+  HttpMethod,
+  httpClient,
+} from '@activepieces/pieces-common';
+
+import { ingAustraliaAuth } from '../..';
+import { launch } from 'puppeteer';
+import { login } from 'ing-au-login';
+
+export const downloadTransactions = createAction({
+  // auth: check https://www.activepieces.com/docs/developers/piece-reference/authentication,
+  name: 'downloadTransactions',
+  auth: ingAustraliaAuth,
+  displayName: 'Download Transactions',
+  description:
+    'Download bank transactions in multiple formats, such as CSV or OFX.',
+  props: {
+    account_number: Property.ShortText({
+      displayName: 'Account Number',
+      required: true,
+    }),
+  },
+  async run(context) {
+    // Log into the ING web portal, using the browser.
+    const browser = await launch({
+      headless: true,
+      args: ['--no-sandbox', '--disable-setuid-sandbox'],
+    });
+    const page = await browser.newPage();
+    const authToken = await login(
+      page,
+      context.auth.username,
+      context.auth.password
+    );
+    await page.close();
+    await browser.close();
```

Comment:
TODO: Move this into its own function. Need to make sending API requests more streamlined. Also need to cache this for a bit. Need to ensure we close the browser process, always. Got to catch exceptions.